vertx-web/src/main/java/io/vertx/ext/web/handler/impl/StaticHandlerImpl.java

```diff
@@ -74,17 +96,40 @@
   private final FSTune tune = new FSTune();
   private final FSPropsCache cache = new FSPropsCache();
 
+  /**
+   * Constructor called by static factory method
+   * 
+   * @param visibility          path specified by root is RELATIVE or ROOT
+   * @param staticRootDirectory path on host with static file location
+   */
+  public StaticHandlerImpl(FileSystemAccess visibility, String staticRootDirectory) {
+
+    this.allowRootFileSystemAccess = FileSystemAccess.ROOT.equals(visibility);
```

Comment:
I think for readablility we can avoid the `.equals()` because enums can be compared by reference, however this could lead to problems in the future if, say, we add another entry to the enum, e.g.: `classpath`.

It may look a bit more verbose but adding a switch:

```java
switch (visibility) {
  case ROOT:
    allowRootFileSystemAccess = true;
    break;
 case RELATIVE:
   allowRootFileSystemAccess = false;
   break;
 default:
   throw new IllegalStateException("Unsupported visibility: " + visibility);
}
```

would be the safest imho

Reply:
updated switch